### PR TITLE
Github workflow added

### PR DESCRIPTION
**Pull request by antongospod**
Runner will run on PR and Push to dev|main branch.

**CI is divided into several stages:**
- Linting
- Type Check
- Tests

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+      - name: Type check 🔤
+        run: npm run type-check
+
+  test:
+    runs-on: ubuntu-latest
+
+    timeout-minutes: 5
+
+    steps:
```
Comment (nelkor) [inline .github/workflows/ci.yaml:51]: The indents look redundant. Let's write beautifully even in configuration files)

```suggestion
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
```

**Review by nelkor — CHANGES_REQUESTED** [review]
Everything is cool, except for some little things. Btw, what does "v3" mean in "actions/checkout@v3"?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        uses: actions/checkout@v3
+
+      - name: Install 👨🏻‍💻
+        run: npm i
+
+      - name: Lint 👀
+        run: npm run lint
+
+      - name: Prettier 🪞
+        run: npm run format:fix
```
[the remark this answers is not in the document]
Comment (antongospod) [reply .github/workflows/ci.yaml:32]: This action checks-out your repository under $GITHUB_WORKSPACE, so your workflow can access it. v3 indicates that we are using the third version of this library. If not specified, the main branch will be used, not all changes in this branch may make it into the final release. The examples in the documentation use version 3. (https://github.com/actions/checkout)
 Yes, you're right. `npm run format:fix` a mistake.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - dev
+      - main
```
[the remark this answers is not in the document]
Comment (antongospod) [reply .github/workflows/ci.yaml:7]: Once merged with the dev branch, the checks will be carried out again. If you are sure that no one will ever flush directly without a PR, this section can be deleted

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        run: npm i
+
+      - name: Type check 🔤
+        run: npm run type-check
+
+  test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
```
Comment (nelkor) [inline .github/workflows/ci.yaml:45]: Look, in all places except this one there is no indent before the steps key.

```suggestion
    steps:
```

**Review by nelkor — CHANGES_REQUESTED** [review]
Just one small edit and it will be approved)

**Review by nelkor — APPROVED** [review]
Well done @antongospod!

It remains to add these changes to the repository and check how GitHub actions will work)

**Pull request merged**